main.tf

```diff
@@ -256,6 +256,10 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = local.automatic_channel_upgrade_check
       error_message = "Either disable automatic upgrades, or only specify up to the minor version when using `automatic_channel_upgrade=patch` or don't specify `kubernetes_version` at all when using `automatic_channel_upgrade=stable|rapid|node-image`. With automatic upgrades `orchestrator_version` must be set to `null`."
     }
+    precondition {
+      condition     = var.role_based_access_control_enabled || (!var.role_based_access_control_enabled && !var.rbac_aad)
```

Comment:
@lonegunmanb could you please review if this precondition looks good to you ? thanks

This is the goal:

| role_based_access_control_enabled | rbac_aad | condition |
| --- | --- | --- |
|True| True | True |
|True| False | True |
|False| True | False |
|False| False | True |

Reply:
I think we can simplify this condition to:

```hcl
condition     = var.role_based_access_control_enabled || !var.rbac_aad
```